### PR TITLE
storage/engine: minor renaming to clarify semantics

### PR DESCRIPTION
**Pull request by petermattis**
Rename `rocksDBBatchIterator` to `batchIterator` and remove the
semantics that a `batchIterator` is tied to the lifetime of a batch. Add
`reusableBatchIterator` which ties the iterator to the lifetime of the
batch. Remove `disposableIterator` which has been replaced by
`batchIterator`.

Release note: None

### CODE REVIEW

**Comment by petermattis** [comment]
No functional changes. I think this naming is a bit clearer. Let me know if you disagree or have another suggestion.

File: pkg/storage/engine/rocksdb.go [hunk 1]
```diff
@@ -1276,78 +1276,79 @@ func (r *distinctBatch) close() {
...
+// in a batch are flushed before performing read operations.
+type batchIterator struct {
 	iter  rocksDBIterator
 	batch *rocksDBBatch
 }
 
-func (r *rocksDBBatchIterator) Close() {
-	// rocksDBBatchIterator.Close() leaves the underlying rocksdb iterator open
+func (r *batchIterator) Close() {
+	// batchIterator.Close() leaves the underlying rocksdb iterator open
```
Comment (tschottdorf) [inline pkg/storage/engine/rocksdb.go]: Isn't that comment wrong now?

File: pkg/storage/engine/rocksdb.go [hunk 2]
```diff
@@ -1358,44 +1359,47 @@ func (r *rocksDBBatchIterator) MVCCScan(
...
 	return r.iter.iter
 }
 
-// disposableIterator wraps rocksDBBatchIterator and allows Close to free up
-// the associated resources instead of tying them to the lifetime of the
-// batch. See reusableIterator and rocksDBBatch.NewTimeBoundIterator.
-type disposableIterator struct {
-	rocksDBBatchIterator
+// reusableBatchIterator wraps batchIterator and allows reuse of an iterator
+// for the lifetime of a batch.
```
Comment (tschottdorf) [inline pkg/storage/engine/rocksdb.go]: Perhaps a better comment here is

> reusableBatchIterator wraps batchIterator and makes the `Close` method a no-op to allow reuse for the lifetime of the batch. The batch must call `iter.destroy()` when it closes itself.

**Review by tbg — APPROVED** [review]

File: pkg/storage/engine/rocksdb.go (hunk 1 above)
Comment (petermattis) [reply pkg/storage/engine/rocksdb.go]: Yep. Removed. Thanks.

File: pkg/storage/engine/rocksdb.go (hunk 2 above)
Comment (petermattis) [reply pkg/storage/engine/rocksdb.go]: Agreed. Done.

**Pull request merged**